File: 2015/02/Part1.py
def paper(s: str) -> int:
    l,w,h = s.split("x")
    l=int(l)
    w=int(w)
    h=int(h)
    return 2*(l*w + w*h + h*l) + min(l*w, w*h, h*l)

with open("input.txt") as f:
    print(sum(paper(line) for line in f))

File: 2023/01/Part1.ts
import { readFileSync } from 'fs';

function getNum(line: string): number {
    let groups = line.match(/^\D*(?<digit>\d)/)?.groups;
    if (groups === undefined) {
        throw new Error("No first digit. Line: '" + line + "'");
    }
    let digit1: string = groups.digit;

    groups = line.match(/(?<digit>\d)\D*$/)?.groups;
    if (groups === undefined) {
        throw new Error("No last digit. Line: '" + line + "'");
    }
    let digit2: string = groups.digit;

    let num: string = digit1 + digit2;
    return Number(num);
}

let total: number = 0;
for (let line of readFileSync("input.txt", "utf-8").split("\n")) {
    if (line.length > 0) {
        total += getNum(line);
    }
}
console.log(total);
